documentation_quality_analysis/analyze_library/prioritize_api/llm/discusses_api_pipeline.py

```diff
@@ -0,0 +1,71 @@
+import json
+import os
+from string import Template
+
+import pandas as pd
+from bs4 import BeautifulSoup
+
+from analyze_library.prioritize_api.heurictics.has_api_analysis import get_all_posts
+from analyze_library.prioritize_api.llm.detect_api_discussion import detect_api_discussion
+
+ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
+TEMPLATE_PATH = os.path.join(ROOT_DIR, "templates")
+DATA_PATH = os.path.join(ROOT_DIR, "data")
+
+
+def get_prompt_messages(library, api_fqn, post_body, mode='one_shot'):
+    metadata = {
+        'library': library,
+        'api_fqn': api_fqn,
+        'post_body': post_body
+    }
+
+    with open(os.path.join(DATA_PATH, "discusses_api_example.txt")) as f:
+        metadata['learning_example'] = f.read()
+
+    if mode == 'one_shot':
+        with open(os.path.join(TEMPLATE_PATH, "few_shot_learning.json")) as f:
+            messages = json.load(f)
+
+            messages['messages'][-1]['content'] = Template(messages['messages'][-1]['content']).substitute(metadata)
```

Comment:
this is a bit confusing, why the -1 here?

Reply:
In the template, the first few message contents are examples and hard coded, only the last item of the list has the placeholders for my new post, that's why. I will change the template as we talked so this will become less confusing.